core/src/main/scala/pink/cozydev/lucille/QueryParser.scala

```diff
@@ -232,12 +226,19 @@ private object Parser {
     Fuzzy(q, n)
   }
 
-  /** Parse a prefix term query
-    * e.g. 'jump*'
+  /** Parse term queries, prefix queries, and wildcards
+    * e.g. 'cat', 'catch22', 'catch*', 'c?tch'
     */
-  val prefixT: P[Prefix] =
-    (term.soft <* P.char('*'))
-      .map(Prefix.apply)
+  val termAndWildCardQ: P[Query] = {
+    val single = P.char('?').as(WildCardOp.SingleChar)
+    val many = P.char('*').as(WildCardOp.ManyChar)
+    val str = term.map(s => WildCardOp.Str(s))
+    (single | many | str).rep.map {
+      case NonEmptyList(WildCardOp.Str(term), Nil) => Term(term)
+      case NonEmptyList(WildCardOp.Str(prfx), WildCardOp.ManyChar :: Nil) => Prefix(prfx)
+      case termOps => WildCard(termOps)
```

Comment:
This is perhaps particularly important to call out, we've combined a couple parsers here.
Previously we had `P[Term]` and `P[Prefix]` separate.
Now they are combined into one `P[Query]`.
This is because a `Wildcard` could technically be just a regular string, or it could be a prefix.
So we construct them all as `Wildcard` and then examine it to see if it matches a simpler case we already have a query node for, and then return that.